### ISSUE TITLE
[Bug]: multi-cn e2e bvt failed

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#6423]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): zhangxu19830126:add_multi_cn_tcp_proxy
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

bvt pass rate only 90%, u can find log here: [multi-cn-e2e_linux_x86_reports (2).zip](https://github.com/matrixorigin/matrixone/files/9963447/multi-cn-e2e_linux_x86_reports.2.zip)

many errors, like
```
internal error: panic runtime error: index out of range [2] with length 2: 
runtime.goPanicIndex
	/opt/hostedtoolcache/go/1.19.2/x64/src/runtime/panic.go:113
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).generateNodes
	/home/runner/work/matrixone/matrixone/head/pkg/sql/compile/compile.go:1277
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).compileTableScan
	/home/runner/work/matrixone/matrixone/head/pkg/sql/compile/compile.go:602
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).compilePlanScope
	/home/runner/work/matrixone/matrixone/head/pkg/sql/compile/compile.go:358
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).compilePlanScope
	/home/runner/work/matrixone/matrixone/head/pkg/sql/compile/compile.go:401
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).compilePlanScope
	/home/runner/work/matrixone/matrixone/head/pkg/sql/compile/compile.go:375
```

### Steps to Reproduce

```Markdown
https://github.com/matrixorigin/matrixone/actions/runs/3420985405
```

**Comment by sukki37** [comment]
@aressu1985 Please pay attention on this, the logs of the mo-tester here are also very strange

**Comment by jensenojs** [comment]
Now there is only one test/cases/expression/cte.sql that has not passed, so let's take a look

**Comment by fengttt** [comment]
https://github.com/matrixorigin/matrixone/commit/dcdfb3510e1b17b60bc2326f7a8d61f866b2be38

Reverted.

**Comment by fengttt** [comment]
Keep this one at S-1, because we believe this PR is absolutely necessary.

**Comment by m-schen** [comment]
I have fix the reverted-pr's bug. and how to verify this ? maybe that's done.

**Comment by sukki37** [comment]
> I have fixed the reverted-PR's bug. and how to verify this? maybe that's done.

Run `make ci LAUNCH=launch-tae-multi-CN-tae-DN` repeatedly.

And I will keep an eye on the results of multi-cn e2e, and if it does not appear in the near future, I will downgrade this issue.

**Comment by aressu1985** [comment]
still has problem

**Comment by heni02** [comment]
temporary Please @sukki37  update the test results of yesterday

**Comment by sukki37** [comment]
waiting for #6529

**Comment by sukki37** [comment]
Not repro in the recent commits; @aressu1985, please close this issue if it passes your test.

**Comment by aressu1985** [comment]
fixed: 2f62c7d7c082f91416190e882ae9c7e30cbb8855
